src/MacroTools/Powers/Transfiguration.cs

```diff
@@ -0,0 +1,67 @@
+﻿using System;
+using MacroTools.Extensions;
+using MacroTools.FactionSystem;
+using Microsoft.VisualBasic;
+using WCSharp.Events;
+using WCSharp.Shared.Data;
+using static War3Api.Common;
+
+namespace MacroTools.Powers
+{
+  /// <summary>
+  /// Causes dying units to Transfiguring at a specific location.
+  /// </summary>
+  public sealed class Transfiguration : Power
+  {
+    private readonly float _chance;
+    private readonly Point _returnPoint;
+    private readonly Rectangle _noReturnRect;
+    public int AnimatedArmorID { get; init; } = FourCC("o01I");
+
+    /// <summary>
+    /// The condition that units need to pass to be eligible for rematerialization.
+    /// </summary>
+    public Func<unit, bool> EligibilityCondition { get; init; } = _ => true;
+
+    /// <summary>
+    /// Initializes a new instance of the <see cref="Transfiguring"/> class.
+    /// </summary>
+    /// <param name="chance">The chance that dying units have to rematerialize.</param>
+    /// <param name="returnPoint">Where rematerialized units appear.</param>
+    /// <param name="returnPointName">A user-friendly name for where rematerialized units appear.</param>
+    /// <param name="noReturnRect">Units that die within this area are not rematerialized.</param>
+    public Transfiguration(float chance, Point returnPoint, string returnPointName, Rectangle noReturnRect)
+    {
+      _chance = chance;
+      _returnPoint = returnPoint;
+      _noReturnRect = noReturnRect;
+      Description = $"Your non-Resistant units have a {chance*100}% chance to transfigure as an animated armor in {returnPointName} upon death.";
```

Comment:
```suggestion
      Description = $"Your non-Resistant units have a {chance*100}% chance to be transfigured into as an animated armor in {returnPointName} upon death.";
```
"Transfigure" as I've seen it is typically phrased as an action that happens to something rather than a process that thing undertakes of its own accord.